src/Repository/AssessorRequestRepository.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace AppBundle\Repository;
+
+use AppBundle\Assessor\Filter\AssessorRequestFilters;
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\AssessorRequest;
+use AppBundle\Entity\VotePlace;
+use Doctrine\ORM\QueryBuilder;
+use Symfony\Bridge\Doctrine\RegistryInterface;
+
+class AssessorRequestRepository extends AbstractAssessorRepository
+{
+    private const ALIAS = 'ar';
+
+    public function __construct(RegistryInterface $registry)
+    {
+        parent::__construct($registry, AssessorRequest::class);
+    }
+
+    public function findMatchingRequests(Adherent $manager, AssessorRequestFilters $filters): array
+    {
+        if (!$manager->isAssessorManager()) {
+            return [];
+        }
+
+        $qb = $this->createQueryBuilder(self::ALIAS);
+
+        $filters->apply($qb, self::ALIAS);
+
+        self::addAndWhereManagedBy($qb, $manager, self::ALIAS);
+
+        $requests = $qb->getQuery()->getResult();
+
+        if ($filters->isStatusUnprocessed()) {
+            return $this->findAssessorRequests($requests);
+        }
+
+        foreach ($requests as $k => $request) {
+            $requests[$k] = [
+                'data' => $request,
+                'matchingVotePlacesCount' => 1,
+            ];
+        }
+
+        return $requests;
+    }
+
+    private function findAssessorRequests(array $assessorRequests): array
+    {
+        if (!\count($assessorRequests)) {
+            return [];
+        }
+
+        $alias = VotePlaceRepository::ALIAS;
+
+        $qb = $this->_em->createQueryBuilder()
+            ->select('COUNT(DISTINCT '.$alias.'.id)')
+            ->from(VotePlace::class, $alias)
+        ;
+
+        foreach ($assessorRequests as $key => $assessorRequest) {
+            $votePlacesCountQuery = clone $qb;
+
+            self::addAndWherePostalCodeFindInSet($votePlacesCountQuery, $assessorRequest, $alias);
+            VotePlaceRepository::addAndWhereOfficeAvailability($votePlacesCountQuery, $assessorRequest);
+            VotePlaceRepository::addAndWhereCity($votePlacesCountQuery, $assessorRequest);
+
+            $assessorRequests[$key] = [
+                'data' => $assessorRequest,
+                'matchingVotePlacesCount' => (int) $votePlacesCountQuery->getQuery()->getSingleScalarResult(),
+            ];
+        }
+
+        return $assessorRequests;
+    }
+
+    public function countMatchingRequests(Adherent $manager, AssessorRequestFilters $filters): int
+    {
+        if (!$manager->isAssessorManager()) {
+            return 0;
+        }
+
+        $qb = $this->createQueryBuilder(self::ALIAS);
+
+        $filters->apply($qb, self::ALIAS);
+
+        self::addAndWhereManagedBy($qb, $manager);
+
+        return (int) $qb
+            ->select('COUNT(DISTINCT '.self::ALIAS.'.id)')
+            ->getQuery()
+            ->getSingleScalarResult()
+        ;
+    }
+
+    public function isManagedBy(Adherent $manager, AssessorRequest $assessorRequest): bool
```

Comment:
I think we can do this job in PHP without call our DB, wdyt?

Reply:
done :)